### DISCUSSION TITLE
Framework evaluation for my company

### DISCUSSION THREAD

**Discussion by maxmontgmx** [oqtane/oqtane.framework#3063] [General]
Hi all,
I'm evaluating this framework to host a lot of small apps (developed in legacy NET framework), developed internally for internal usages.
Theses small apps need access to hardware ressources (network, serial port, printer, ...) on local network or LAN.
With Blazor client side, I believed that these techno cover my needs. but it's not the case because WebAssembly doesn't have all features (like Net Socket, Serial port, ...)
Can you confirm that I'm saying or not?
If I'm right, do you have an idea about which solution I can implement to cover my needs?
Thanks in advance for your help

**Comment by sbwalker** [comment]
I assume you are replacing legacy desktop apps? If so, the Blazor Hybrid / .NET MAUI hosting model may provide what you are looking for. You can build Oqtane modules which run inside of a MAUI client - see  https://www.youtube.com/watch?v=xyus7zlmgdc Use the Oqtane.Maui solution file in the root folder.

**Reply by maxmontgmx** [reply]
Thanks for response.
I dont exactly know what a Maui client Is.
Sorry i m juste discovering blazer/Maui/razor techno.
I will look at octane.maui solution tomorrow.
I Guess it IS a multi plateform app which Can Host octane web site pages and modules

**Reply by maxmontgmx** [reply]
Hi,
I tried MAUI client and it's working with http://localhost:{port}
But when I try my second web site by modifying MauiProgram.cs static string apiurl = "http://localhost:{port}/prod/";
it displays the main web site (http://localhost:{port})
What is the cause of this issue?

Also, when I setup web site Hosting Model as "Blazor Hybrid", Web Browser displays only "Site Is Disabled"
But in this case I don't know coming back to "Blazor Server" value.

Thanks for your help
